src/source-updater.js

```diff
@@ -41,6 +54,16 @@ export default class SourceUpdater {
       this.runCallback_();
     };
 
+    // Fixes https://github.com/videojs/videojs-contrib-hls/issues/963
+    // We run this on the next tick as it seems
+    // that sometimes when the `open` event is triggered, or when
+    // the readyState indicates open, the
+    // MediaSource is actually not ready yet to accept
+    // calls to its `addSourceBuffer` method.
+    // However when state is settled on the next tick,
+    // it is safe to do so.
+    const createSourceBufferDeferred = () => setTimeout(createSourceBuffer, 50);
```

Comment:
running this immediately (zero timeout) doesn't fix the issue.

we have experienced that this type of "delay" allows for a reliable behavior.

that proves this is a pretty "fuzzy" issue in the lower layers imho.

Reply:
maybe there is a special event when this should be happen? because the 50ms seems random.